src/Pages/Landing/Landing.js

```diff
@@ -0,0 +1,57 @@
+import * as React from "react";
+import { Card, ListGroup, Button } from "react-bootstrap";
+import { useQuery } from "@apollo/react-hooks";
+import "./Landing.scss";
+import { BsFillInfoSquareFill as IconInfo } from "react-icons/bs";
+import GET_USERDATA from './../../Apollo/Queries/User'
+import { useHistory } from "react-router-dom";
+
+function Landing() {
+  
+  const { loading, error, data } = useQuery(GET_USERDATA);
+  let history = useHistory();
+
+  const handleOnClick = (e) => {
+    e.preventDefault();
+    history.push("/")
+  };
+
+  if (loading) return <p>Loading...</p>;
+  if (error) return <p>{error}</p>;
+  if (data) {
+    const { id, username, firstname, lastname, password } = data.currentUser;
+    return (
+      <div className="containerCard">
+        <Card>
+          <Card.Header>
+            Datos Usuario Ingresado <IconInfo></IconInfo>
+          </Card.Header>
+          <ListGroup className="cardContent">
+            <ListGroup.Item>
+              ID : {id ? <p>{id}</p> : <p>no tiene id</p>}
+            </ListGroup.Item>
+            <ListGroup.Item>
+              User Name :
+              {username ? <p>{username}</p> : <p>no tiene username</p>}
+            </ListGroup.Item>
+            <ListGroup.Item>
+              First Name :{" "}
+              {firstname ? <p>{firstname}</p> : <p>no tiene firstname</p>}
+            </ListGroup.Item>
+            <ListGroup.Item>
+              Last Name :{" "}
+              {lastname ? <p>{lastname}</p> : <p>no tiene lastname</p>}
+            </ListGroup.Item>
+            <ListGroup.Item>
+              Password : <p className="password">{password}</p>
+            </ListGroup.Item>
+            <ListGroup.Item >
+              <Button className="buttonLanding" onClick={handleOnClick}>Logout</Button>
+            </ListGroup.Item>
+          </ListGroup>
+        </Card>
+      </div>
+    );
+  }
```

Comment:
The problem with this is that, if there is no `loading`, `error` and `data` values, your component is not returning anything by default. The last `return` statement should not be inside an `if`. I suggest:

```suggestion
  if (loading) return <p>Loading...</p>;
  if (error) return <p>{error}</p>;

    return (
      <div className="container">
        {data &&
          <Card
            currentUser={data.currentUser}
            onLogoutButtonClick={onLogoutClick}
          />
        }
      </div>
    )
```
Then you can extract the rest of the code into a Card component. Notice how if there is no `data` you do not render a Card.

Reply:
😮 that's certainly true! I didn't think about that path before.